crates/sui-json-rpc/src/coin_api.rs

```diff
@@ -832,9 +749,9 @@ mod tests {
             assert!(response.is_err());
             let error_result = response.unwrap_err();
             let error_object: ErrorObjectOwned = error_result.into();
-            let expected = expect!["-32000"];
+            let expected = expect!["-32001"];
             expected.assert_eq(&error_object.code().to_string());
-            let expected = expect!["Storage error"];
+            let expected = expect!["Unexpected error encountered while reading state."];
```

Comment:
are there expected errors that occur? Could probably just say "An error occurred..."

Reply:
Gotcha, that's fine too. Expected errors that occur would be client errors, so perhaps instead of `Unexpected` I can label the variant `Internal`? imo the details shouldn't be exposed on the rpc, but it'll still be logged in debug format. So the log will contain the actual error regarding what occurred. To the rpc caller, meanwhile, these `Unexpected` errors are not things they can act on, other than just retrying the call again.

Reply:
I think `Internal` or `Server` error makes better sense than `Unexpected`.